src/common/database.c

```diff
@@ -1371,6 +1371,30 @@ static gboolean pid_is_alive(int pid)
   }
 #else
   pid_is_alive = !((kill(pid, 0) == -1) && errno == ESRCH);
+
+#ifdef __linux__
+  // If this is Linux, we can query /proc to see if the pid is
+  // actually a darktable instance.
+  if (pid_is_alive)
+  {
+    char filename[64];
```

Comment:
pid_t is 32bit int.

64 should be sufficient: strlen("/proc/[int64_max]/cmdline") < 64.